src/Writer.php

```diff
@@ -68,18 +78,87 @@ public function testConnection(): void
 
     public function write(): void
     {
-        // TODO
+        // Create CSV reader
+        $this->logger->info(sprintf(
+            'Exporting table "%s" in "%s" mode ...',
+            $this->config->getTableId(),
+            $this->config->getMode()
+        ));
+
+        // Create mapper
+        $mapper = $this->messageMapperFactory->create();
+
+        // Register a new NodeJs process to event loop.
+        $process = $this->createNodeJsProcess('write.js');
+        $messageStream = $process->getMessageStream();
+        $messageWriter = new MessageWriter($messageStream);
+
+        // Throw an exception on process failure
+        $process
+            ->getPromise()
+            ->done(null, function (Throwable $e): void {
+                if ($e instanceof ProcessException && $e->getExitCode() === 1) {
+                    throw new UserException('Export failed.', $e->getCode(), $e);
+                } else {
+                    throw new ApplicationException($e->getMessage(), $e->getCode(), $e);
+                }
+            });
+
+        // Schedule the first batch
+        $this->futureWriteCsvRows($mapper->getMessages(), $messageWriter);
+
+        // Start event loop
+        $this->loop->run();
+
+        // Done
+        $this->logger->info(sprintf(
+            'Exported all %d rows from the table "%s".',
+            $messageWriter->getCount(),
+            $this->config->getTableId()
+        ));
+    }
+
+    protected function writeCsvRows(Iterator $messages, MessageWriter $messageWriter): void
+    {
+        if ($messageWriter->isBufferFull()) {
+            usleep(5000); // wait 5ms and check again
+            return;
+        }
+
+        for ($i = 0; $i < self::CSV_ROWS_BATCH_SIZE && !$messageWriter->isBufferFull(); $i++) {
+            if (!$messages->valid()) {
+                // No more rows
+                $messageWriter->finish();
+                return;
+            }
+
+            $messageWriter->writeMessage($messages->current());
+            $messages->next();
+        }
+    }
+
+    protected function futureWriteCsvRows(Iterator $messages, MessageWriter $messageWriter): void
+    {
+        // We write CSV lines in batches to the NodeJs file descriptor.
+        // After each batch, execution returns to the loop, so that it can process other events.
+        // Event loop then call "futureTick" callback.
+        $this->loop->futureTick(function () use ($messages, $messageWriter): void {
```

Comment:
Kniznica Keboola CSV je synchronna, takze aby ju bolo mozne pouzit tak to pracuje takto:
- Zapise sa 10 riadkov.
- Potom sa riadenie odovzda event loop.
- Event loop vyriesi streamy a ostatne async veci.
- A kedze sme zaregistrovali callback s `futureTick` ... tak sa potom zapise opat 10 riadkov, a takto to ide, ... az kym nie je vsetko zapisane, a preto sa uz dalsi `futureTick` callback neregistruje.